lib/include/hse_ikvdb/sched_sts.h

```diff
@@ -1,69 +1,81 @@
 /* SPDX-License-Identifier: Apache-2.0 */
 /*
- * Copyright (C) 2015-2021 Micron Technology, Inc.  All rights reserved.
+ * Copyright (C) 2015-2022 Micron Technology, Inc.  All rights reserved.
  */
 
 #ifndef HSE_IKVDB_STS_H
 #define HSE_IKVDB_STS_H
 
 #include <hse_util/platform.h>
 #include <hse_util/workqueue.h>
+#include <hse_util/list.h>
 
 /* MTF_MOCK_DECL(sched_sts) */
 
 struct sts;
 struct sts_job;
-struct kvdb_rparams;
 
 typedef void sts_job_fn(struct sts_job *job);
-typedef void sts_cancel_fn(struct sts_job *job);
+typedef int sts_print_fn(struct sts_job *job, char *buf, size_t bufsz, const char *fmt);
 
 /**
  * struct sts_job - short term scheduler job handle
- * @sj_link:      for keeping jobs on a linked list (scheduler internal use)
- * @sj_sts:       handle to scheduler that manages this job
+ * @sj_runq_link: run queue list linkage
+ * @sj_print_fn:  callack to print job status line
  * @sj_job_fn:    job handler, set by client, invoked by scheduler
- * @sj_cancel_fn: job cancel function, set by client, invoked by scheduler
- * @sj_tag:       tag, set by client, used to cancel all jobs with same tag
- * @sj_qnum:      scheduler queue
+ * @sj_id:        job ID
+ * @sj_status:    job status code
+ *
+ * A job is initialized with status code 'I' (idle) and then set to 'R'
+ * (run) just prior to invoking the user job function.  The callee may
+ * then call sts_job_status_set() at any time to set the status code
+ * to indicate the current status of the job.
  */
 struct sts_job {
+    struct list_head   sj_runq_link;
+    sts_print_fn      *sj_print_fn;
     sts_job_fn        *sj_job_fn;
-    sts_cancel_fn     *sj_cancel_fn;
-    u64                sj_tag;
-    uint               sj_qnum;
     uint               sj_id;
+    char               sj_status;
     struct work_struct sj_work;
 };
 
 /**
  * sts_create() - create a short term scheduler for kvdb compaction work
- * @rp:      kvdb run-time parameters
  * @name:    name
  * @nq:      number of queues
  * @sts:     (out) short term scheduler handle
  */
 /* MTF_MOCK */
 merr_t
-sts_create(struct kvdb_rparams *rp, const char *name, uint nq, struct sts **sts);
+sts_create(const char *name, uint nq, struct sts **sts);
```

Comment:
Would be nice to stop using `uint` at some point. Seems like POSIX doesn't even guarantee these shorthands will exist https://pubs.opengroup.org/onlinepubs/009696899/basedefs/sys/types.h.html.

Reply:
Right, I think u_int and u_long are sanction, IIRC...  Lemme look at that page..

Reply:
u_int seems to be fully embraced by Linux (apparently it's a BSDism), and we've pretty much painted ourselves into being stuck on Linux w/o some very serious refactoring, so maybe we should start using u_int et. al.   But if we decide it's a worthwhile endeavor then we should just do it all at one in one giant commit, IMO..  I rather like uint/ulong/..., and if they aren't available on a platform it's no biggie to create our own..

Reply:
To me neither buys anything over just writing out `unsigned long` 😃